code/modules/spells/spell_types/lightning.dm

```diff
@@ -65,22 +65,47 @@
 /obj/effect/proc_holder/spell/targeted/tesla/proc/Bolt(mob/origin,mob/target,bolt_energy,bounces,mob/user = usr)
 	origin.Beam(target,icon_state="lightning[rand(1,12)]",time=5)
 	var/mob/living/carbon/current = target
-	if(current.anti_magic_check())
+	var/whywouldyoueverusethisagainstajedi = FALSE
+	var/helditem=current.get_active_held_item()
+	if(helditem)
+		if(istype(helditem, /obj/item/melee/transforming/energy/sword) && helditem.active) //only works if they have an esword in their active hand
+			whywouldyoueverusethisagainstajedi = TRUE
+		else if(istype(helditem, /obj/item/twohanded/dualsaber) && helditem.wielded) //or a double-bladed esword
+			whywouldyoueverusethisagainstajedi = TRUE
+		else if(istype(helditem, /obj/item/nullrod/claymore/saber)) //or one of the null rod variants that look like eswords
+			whywouldyoueverusethisagainstajedi = TRUE
```

Comment:
all of this is bad and can be done in a much better way (first thought is a typecached list)

Reply:
Dont forget your spaces as well to make the code look cleaner.